lib/launchers/amz.rb

```diff
@@ -47,9 +47,22 @@ def initialize(access_key: nil, secret_key: nil, service_name: nil)
           awscred["AWSSecretKey"]
         )
       }) )
-      client = Aws::EC2::Client.new
-      @ec2 = Aws::EC2::Resource.new(client: client)
+    end
+
+    private def ec2
+      @ec2 ||= Aws::EC2::Resource.new(client: ec2_client)
+    end
+
+    private def ec2_client
+      @clien ||= Aws::EC2::Client.new
```

Comment:
@akostadinov is this used anywhere?  Also why named it `@clien` instead of `@client`

Reply:
it must be `@client`. Typo. The `ec2_client` **is** used. the variable is for caching the object instead of creating many times.

Reply:
Fixed now, thanks a lot for catching this. Actually bigger issue was that bot s3 and ec2 used the same caching variable :/

Reply:
Fixed in both pull requests.

Reply:
cool, thanks.  the rest looks good!